dotnet/src/Connectors/Connectors.Memory.PgVector/PostgresServiceCollectionExtensions.cs

```diff
@@ -1,163 +1,295 @@
+    /// <param name="lifetime">The service lifetime for the store. It needs to match <see cref="NpgsqlDataSource"/> lifetime. Defaults to <see cref="ServiceLifetime.Singleton"/>.</param>
     /// <returns>Service collection.</returns>
-    public static IServiceCollection AddPostgresVectorStoreRecordCollection<TKey, TRecord>(
+    public static IServiceCollection AddPostgresCollection<TKey, TRecord>(
         this IServiceCollection services,
         string collectionName,
         PostgresCollectionOptions? options = default,
-        string? serviceId = default)
+        ServiceLifetime lifetime = ServiceLifetime.Singleton)
         where TKey : notnull
         where TRecord : class
     {
-        services.AddKeyedTransient<VectorStoreCollection<TKey, TRecord>>(
-            serviceId,
-            (sp, obj) =>
-            {
-                var dataSource = sp.GetRequiredService<NpgsqlDataSource>();
-                options ??= sp.GetService<PostgresCollectionOptions>() ?? new()
-                {
-                    EmbeddingGenerator = sp.GetService<IEmbeddingGenerator>()
-                };
-
-                // The data source has been solved from the DI container, so we do not own it.
-                return (new PostgresCollection<TKey, TRecord>(dataSource, collectionName, ownsDataSource: false, options) as VectorStoreCollection<TKey, TRecord>)!;
-            });
-
-        AddVectorizedSearch<TKey, TRecord>(services, serviceId);
+        Verify.NotNull(services);
+        Verify.NotNullOrWhiteSpace(collectionName);
+
+        services.Add(new ServiceDescriptor(typeof(PostgresCollection<TKey, TRecord>), (sp) =>
+        {
+            var dataSource = sp.GetRequiredService<NpgsqlDataSource>();
+            var copy = GetCollectionOptions(sp, _ => options);
+
+            // The data source has been solved from the DI container, so we do not own it.
+            return new PostgresCollection<TKey, TRecord>(dataSource, collectionName, ownsDataSource: false, copy);
+        }, lifetime));
+
+        AddAbstractions<TKey, TRecord>(services, serviceKey: null, lifetime);
 
         return services;
     }
 
     /// <summary>
-    /// Register a Postgres <see cref="VectorStoreCollection{TKey, TRecord}"/> and <see cref="IVectorSearchable{TRecord}"/> with the specified service ID
-    /// and where the NpgsqlDataSource is constructed using the provided parameters.
+    /// Registers a <see cref="PostgresCollection{TKey, TRecord}"/> as <see cref="VectorStoreCollection{TKey, TRecord}"/>, with the specified connection string and service lifetime.
+    /// </summary>
+    /// <inheritdoc cref="AddKeyedPostgresCollection{TKey, TRecord}(IServiceCollection, object, string, string, PostgresCollectionOptions?, ServiceLifetime)"/>
+    public static IServiceCollection AddPostgresCollection<TKey, TRecord>(
+        this IServiceCollection services,
+        string collectionName,
```

Comment:
I think we should consider inversing the order of the `collectionName` and `connectionString` parameters - the connection string identifies the database, whereas the collection name identifies the table inside that database.

If we want to, we can also shorten `collectionName` to `name` (since the method is `AddPostgresCollection`) but I don't really care much either way.

Reply:
Resolving as we decided against this in the design discusion.